extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeFeedExtractor.java

```diff
@@ -57,19 +61,44 @@ public ListExtractor.InfoItemsPage<StreamInfoItem> getInitialPage() {
     @Nonnull
     @Override
     public String getId() {
-        return document.getElementsByTag("yt:channelId").first().text();
+        final String idFromAuthorUriElement = getUrl().replace(WEBSITE_CHANNEL_BASE_URL, "");
+        if (isNullOrEmpty(idFromAuthorUriElement)) {
+            final Element linkElement = document.select("feed > link[rel*=alternate]")
+                    .first();
+            if (linkElement != null) {
+                final String linkElementChannelId = linkElement.attr("href")
+                        .replace(WEBSITE_CHANNEL_BASE_URL, "");
+                if (!isNullOrEmpty(linkElementChannelId)) {
+                    return linkElementChannelId;
+                }
```

Comment:
This will have the same effect I think, since if linkElementChannelId were null or empty then control would reach the other return which would return null anyway.
```suggestion
                return linkElement.attr("href")
.replace(WEBSITE_CHANNEL_BASE_URL, "");
```

Reply:
I just noticed that this could be used as a fallback for the channel URL, so I am moving this part into the `getUrl` method.